src/settings/JobProfiles/NewJobProfileRoute/NewJobProfileRoute.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isEmpty } from 'lodash';
+
+import { stripesConnect } from '@folio/stripes/core';
+
+import { JobProfilesForm } from '../JobProfilesForm';
+
+const getFormattedMappingProfiles = mappingProfiles => {
+  if (isEmpty(mappingProfiles?.records)) return [];
+
+  return mappingProfiles.records.reduce((memo, current) => {
+    memo.push({
+      label: current.name,
+      value: current.id,
+    });
+
+    return memo;
+  }, []);
+};
+
+const NewJobProfileRoute = props => {
+  const {
+    onCancel,
+    onSubmit,
+    resources: { mappingProfiles },
+  } = props;
+
+  return (
+    <JobProfilesForm
+      onSubmit={onSubmit}
+      hasLoaded={mappingProfiles && mappingProfiles.hasLoaded}
```

Comment:
```suggestion
      hasLoaded={mappingProfiles?.hasLoaded}
```

Reply:
As far as I understand mappingProfiles?.hasLoaded will put undefined, but bool is expected. Do we really need it in this case?

Reply:
I suggest leaving as it is for the next reasons:
1.  It puts undefined instead of boolean
2. It seems it saves just a couple of symbols but the final code (after babel applying) will be even longer. The implemented logic is clear enough.

Reply:
The `mappingProfiles && mappingProfiles.hasLoaded` will also result in `undefined` if `mappingProfiles` is false.

Reply:
@VictorSoroka1  thanks for clarification. It was already replaced

Reply:
Thank you, Your comment is valid as well about the prop validation, we need to pay attention to this as it would result in warnings in console.